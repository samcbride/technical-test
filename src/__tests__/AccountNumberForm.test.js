import React from "react";
import { render } from "@testing-library/react";
import AccountNumber from "../components/AccountNumberForm";

describe("AccountNumberForm", () => {
  //   const validProps = {
  //     accountNumber: 1234,
  //   };

  it("renders correctly", () => {
    const { asFragment } = render(<AccountNumber />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("check button exists with correct text", () => {
    const { getByRole } = render(<AccountNumber />);
    const button = getByRole("button", { name: /Submit/i });
    expect(button).toBeInTheDocument();
  });
});
