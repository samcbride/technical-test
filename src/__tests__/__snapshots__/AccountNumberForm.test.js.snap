// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountNumberForm renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="account-number"
    >
      <input
        class="account-number-area"
        placeholder="Enter account number"
        type="text"
      />
      <button
        class="submit-button"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
